### ISSUE TITLE
@storybook/react-native/scripts/generate.js is breaking with latest storybook

### ISSUE THREAD

**Issue by satheshrgs-rksv** [storybookjs/react-native#660]
my main.ts file looks like this
```
// modules
import { StorybookConfig } from '@storybook/react-native';

// NOTE: if you are updating this file make sure to run `npm run storybook-generate`

const main: StorybookConfig = {
  stories: ['../src/components/**/*.stories.?(ts|tsx|js|jsx)'],
  addons: [
    '@storybook/addon-ondevice-controls',
    '@storybook/addon-ondevice-actions',
    './customAddon/theme',
  ],
};

export default main;
```

After recently upgrading to latest 8.4.3 version, `./customAddon/theme` is being excluded from the `storybook.requires.ts` file. From my analysis, this is happening due to https://github.com/storybookjs/react-native/pull/612/files#diff-1582b6229d57d69f719432d26ae15f723fb028825d8a6eb465c0e2fc796c7be8R54 by @onmotion. Kindly suggest on steps to fix it. This is a local addon and i can't export it

**Comment by dannyhw** [comment]
If you make the file in your custom addon folder be preview.js does it work?

You need either a register file or a preview file otherwise it wouldn't work even before 8.4.3 as far as i can tell

**Comment by satheshrgs-rksv** [comment]
That is also failing

**Comment by satheshrgs-rksv** [comment]
```
Error: Cannot find module './customAddon/theme/preview.js'
Require stack:
- /Users/satheshrgs/workspace/mobile/ui-lib/node_modules/@storybook/react-native/scripts/common.js
- /Users/satheshrgs/workspace/mobile/ui-lib/node_modules/@storybook/react-native/scripts/generate.js
- /Users/satheshrgs/workspace/mobile/ui-lib/node_modules/@storybook/react-native/scripts/get-stories.js
- /Users/satheshrgs/workspace/mobile/ui-lib/node_modules/@storybook/react-native/bin/get-stories.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1140:15)
    at Module._resolveFilename (/Users/satheshrgs/workspace/mobile/ui-lib/node_modules/esbuild-register/dist/node.js:4799:36)
    at Function.resolve (node:internal/modules/helpers:188:19)
    at resolveAddonFile (/Users/satheshrgs/workspace/mobile/ui-lib/node_modules/@storybook/react-native/scripts/common.js:78:15)
    at generate (/Users/satheshrgs/workspace/mobile/ui-lib/node_modules/@storybook/react-native/scripts/generate.js:69:25)
    at Object.<anonymous> (/Users/satheshrgs/workspace/mobile/ui-lib/node_modules/@storybook/react-native/scripts/get-stories.js:5:1)
    at Module._compile (node:internal/modules/cjs/loader:1364:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Module._load (node:internal/modules/cjs/loader:1019:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/satheshrgs/workspace/mobile/ui-lib/node_modules/@storybook/react-native/scripts/common.js',
    '/Users/satheshrgs/workspace/mobile/ui-lib/node_modules/@storybook/react-native/scripts/generate.js',
    '/Users/satheshrgs/workspace/mobile/ui-lib/node_modules/@storybook/react-native/scripts/get-stories.js',
    '/Users/satheshrgs/workspace/mobile/ui-lib/node_modules/@storybook/react-native/bin/get-stories.js'
  ]
}
```

**Comment by dannyhw** [comment]
If you can create a reproduction then im happy to debug too

**Comment by onmotion** [comment]
Can you try to resolve (`path.resolve`) the full path to the module maybe?

**Comment by satheshrgs-rksv** [comment]
`require('path').resolve(__dirname, './customAddon/theme')` this works, but it will also contains the full path and will not be able to commit to git. Sample output in the storybook.requires.ts file will be `import "/Users/satheshrgs/workspace/mobile/ui-lib/.storybook/customAddon/theme/register";`

Manually adding `import "./customAddon/theme/register";` in the autogenerated file `.storybook/storybook.requires.ts` for now works

**Comment by dannyhw** [comment]
@satheshrgs-rksv you don't really need to commit the requires file though since its generated, is there a reason you would want to commit it?

**Comment by satheshrgs-rksv** [comment]
Got it, just noticed that it gets generated while doing `npm start`, I hope it's not required

**Comment by dannyhw** [comment]
Either way I will try to create an example and see if I see a similar problem, it should work for relative paths ideally.

**Comment by dannyhw** [comment]
Hey @satheshrgs-rksv I fixed relative local addons in this version 8.4.4-alpha.0 if you could try it out that would be great 🙏

**Comment by satheshrgs-rksv** [comment]
@dannyhw I have tested and it works, Thanks 🙏

**Comment by dannyhw** [comment]
fixed in 8.4.4
